Redesign Registration Page

nam-t24 opened the pull request:
Design registration page to match figma which uses a multipage design for each section.

nam-t24 committed:
refactor: Redesign registration page to multipage design

Registration page is multipage format like the figma design

nam-t24 committed:
refactor: Improve responsive design for mobile

geeth24 commented on pages/register.tsx:
```diff
@@ -275,65 +281,174 @@ export default function Register() {
+                setRegistrationSection(registrationSection + 1);
+              }}
+            >
+              next page
```
when u click next the previous button overlaps
<img width="267" alt="Screenshot 2023-03-02 at 1 30 19 PM" src="https://user-images.githubusercontent.com/47117192/222532839-2caee4e7-70e7-4a29-a057-d4db8f3f7776.png">

geeth24 replied:
next page button is on the right side of page in figma

nam-t24 replied:
what size screen is this in? I'm not getting this issue

geeth24 replied:
3072x1920

nam-t24 replied:
oops I see it. forgot to change some values after copy pasting

geeth24 commented on pages/register.tsx:
```diff
@@ -275,65 +281,174 @@ export default function Register() {
+                  minutes.
+                  <h2 className="sm:text-2xl text-xl font-semibold mt-5 sm:mb-3 mb-1">General</h2>
+                  <div className="flex flex-col">
+                    {generalQuestions.map((obj, idx) => (
```
borders around the input are lighter in figma.

figma:
<img width="526" alt="Screenshot 2023-03-02 at 1 32 15 PM" src="https://user-images.githubusercontent.com/47117192/222533150-d4f93cc8-6fe3-4022-9eae-37f300fa9db5.png">

urs:
<img width="868" alt="Screenshot 2023-03-02 at 1 33 43 PM" src="https://user-images.githubusercontent.com/47117192/222533397-8c519adf-879d-41e9-a7a8-518ec014cbbd.png">

nam-t24 replied:
Is the lighter text for placeholder text or is it full actual input? I thought it was just for placeholder but we're not using any placeholder text. I'll change the border tho

geeth24 replied:
thats just the placeholder, i think kelly's design was inconsistent there, as she made me do this for the sign in page:
<img width="593" alt="Screenshot 2023-03-02 at 2 03 22 PM" src="https://user-images.githubusercontent.com/47117192/222539648-c48adcdd-4c33-4011-841f-37b1aabb8e39.png">
```html
<input
    className="w-full rounded-md border border-complementary/20 p-2 mb-4"
    value={currentEmail}
    onChange={(e) => setCurrentEmail(e.target.value)}
    type="text"
    name="email"
    autoComplete="email"
    placeholder="Email Address*"
/>
```

nam-t24 replied:
Ignore everything I just commented here I must be dyslexic lol totally misread what you typed

nam-t24 committed:
fix: Small UI Fixes to registration page

Next page button is now on the right side. First name autofill only does first
name, text field border is light

geeth24 commented:
@nam-t24 I think we should not let them skip to the next pages if they didn't answer the question. Cause right now they can submit an empty registration. What do u think

nam-t24 commented:
The submit button shouldn't work and would pop up red messages if there are empty fields

geeth24 commented:
yeah but i think it would be a better user experience if it gave instant feedback.

nam-t24 commented:
True but I think it would be nice being able to see the other questions and come back to fill out missed sections or if someone didn't wanna submit the whole thing atm but can look through the other questions ahead and see. But also I have no qualifications when it comes to UX lol

nam-t24 committed:
fix: Fix school and major datalist loading and reduce school list

geeth24 committed:
refactor(resumeupload): Style Resume Upload Button

geeth24 committed:
Merge branch 'develop' into redesign/Registration

geeth24 committed:
Merge branch 'develop' into redesign/Registration

Spaceface16518 commented on pages/register.tsx:
```diff
@@ -45,11 +49,34 @@ export default function Register() {
 
+  useEffect(() => {
+    //load json data into dropdown list for universities and majors
+    if (document.getElementById('schools') !== null) {
+      for (let school of schools) {
+        let option = document.createElement('option');
+        option.text = school['university'];
+        option.value = school['university'];
+        let select = document.getElementById('schools');
+        select.appendChild(option);
+      }
+    }
+
+    if (document.getElementById('majors') !== null) {
+      for (let major of majors) {
+        let option = document.createElement('option');
+        option.text = major['major'];
+        option.value = major['major'];
+        let select = document.getElementById('majors');
+        select.appendChild(option);
+      }
+    }
```
Is it possible to not use `document.getElementById` here? Unless there is some crazy hack going on here that I don't understand, I feel like this is something that should be done using React features (namely, `useRef`).

nam-t24 replied:
I'm kinda just used to using document mb I'll fix later

nam-t24 replied:
Deleted cuz this isn't used anymore

Spaceface16518 commented on styles/globals.css:
```diff
@@ -120,15 +120,6 @@ a.link {
   right: 4rem;
 }
 
-.registrationTitle:after {
-  content: '';
-  display: block;
-  margin: 0 auto;
-  width: 75%;
-  padding-top: 15px;
-  border-bottom: 2px solid black;
-}
-
```
I think this should be in a CSS module local to either the registration question component or the register page.

Spaceface16518 replied:
If it's because of `:after`, there's support for [`before` and `after` psuedo-elements](https://tailwindcss.com/docs/hover-focus-and-other-states#before-and-after) in tailwind. Maybe use that if it's not too ugly.

nam-t24 replied:
I don't remember what this was used for lol I think this was from an old figma design or something

Spaceface16518 commented on public/schools.json:
Just for record's sake: for what reason are we removing these schools?

nam-t24 replied:
Some schools aren't from the US and the chances of someone from these schools going to hackutd are so low they might as well just put other or fill in a custom school. Some people from the past complained that there were too many schools and it was harder to navigate so removing these schools would help

Spaceface16518 commented on pages/register.tsx:
```diff
@@ -45,11 +49,34 @@ export default function Register() {
     formInitialValues['id'] = user?.id || '';
     formInitialValues['preferredEmail'] = user?.preferredEmail || '';
-    formInitialValues['firstName'] = user?.firstName || '';
+    formInitialValues['firstName'] = user?.firstName?.split(' ')[0] || '';
```
What is this supposed to do? Should we discard first names that are multiple words?
If this instead trying to trim the string, it should use [`trimEnd`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimEnd).

nam-t24 replied:
Google treats the display name option as your full name on your google account. So if someone would sign in with google, it would autofill the first name field to your full name. There were some instances of people from hackutd IX not noticing it when registering so their profile names had their last name on twice. I just had it auto fill to the first name associated with the google account to make it easy. If there's a better idea on autofilling it lmk

Spaceface16518 replied:
Okay, I'm not super familiar with how this value is being seeded, so I'll research and come back to this later.

Spaceface16518 commented on pages/register.tsx:
```diff
@@ -253,65 +282,178 @@ export default function Register() {
+              {/* School Questions */}
+              {registrationSection == 1 && (
+                <section className="bg-white lg:w-3/5 md:w-3/4 w-full min-h-[35rem] mx-auto rounded-2xl md:py-10 py-6 px-8 mb-8 text-[#4C4950]">
+                  Please fill out the following fields. The application should take approximately 5
+                  minutes.
```
lmao is this supposed to be an actual estimate? Probably should mark that as TODO/FIXME so that the feature gets implemented at some point.
If there isn't a plan to make this an actual estimate, it feels weird to include such a specific estimation.

nam-t24 replied:
idk this was just here since the very beginning on figma lol

Spaceface16518 replied:
Okay we can just leave it then.

Spaceface16518 reviewed (dismissed):
Haven't looked at the rendered demo yet to compare it to figma, but I had a couple of code quality things.

Spaceface16518 commented on pages/register.tsx:
@kellyxzhou gave us the go ahead to remove all instances of "The application should take approximately 5 minutes", if you want to do that in this PR.

nam-t24 committed:
chore: Remove unused code

abdullah50053 commented:
Dropdown icon overlaps text:
![image](https://user-images.githubusercontent.com/71054105/233814487-072e5597-dff6-4bb7-9127-514ae5252c75.png)
![image](https://user-images.githubusercontent.com/71054105/233814490-f1b86884-07fb-4d3f-9f2b-78e0ce85612a.png)
![image](https://user-images.githubusercontent.com/71054105/233814465-66561caf-bf28-420d-b066-d51483ead66a.png)
![image](https://user-images.githubusercontent.com/71054105/233814505-0aea6e08-5937-4f2c-9423-832a3d3fcd6a.png)
![image](https://user-images.githubusercontent.com/71054105/233814531-411edc59-a19a-4eea-9ea5-baa2ec32a6b8.png)
![image](https://user-images.githubusercontent.com/71054105/233814538-a553bc39-3e7e-4344-83fb-704e68c174d9.png)

Otherwise lgtm.

nam-t24 committed:
fix(registrationquestion): Fix select dropdown arrow overlap

abdullah50053 committed:
Merge branch 'develop' into redesign/Registration
